src/scripting/lua_widget_attributes.cpp

```diff
@@ -0,0 +1,527 @@
+{
+	gui2::widget* w = wp.get_ptr();
+	if(!w) {
+		ERR_LUA << "widget was deleted\n";
+		return;
+	}
+	gui2::window* wd = w->get_window();
+	if(!wd) {
+		ERR_LUA << "canot find window in eidgte callback\n";
+		return;
+	}
+	luaW_callwidgetcallback(L, w, wd, id);
+}
+
+WIDGET_SETTER("on_modified", lua_index_raw, gui2::widget)
+{
+	gui2::window* wd = w.get_window();
+	if(!wd) {
+		throw std::invalid_argument("the widget has no window assigned");
+	}
+	lua_pushvalue(L, value.index);
+	if (!luaW_setwidgetcallback(L, &w, wd, "on_modified")) {
+		connect_signal_notify_modified(w, std::bind(&dialog_callback, L, lua_ptr<gui2::widget>(w), "callback"));
+	}
+}
+
+WIDGET_SETTER("on_left_click", lua_index_raw, gui2::widget)
+{
+	gui2::window* wd = w.get_window();
+	if(!wd) {
+		throw std::invalid_argument("the widget has no window assigned");
+	}
+	lua_pushvalue(L, value.index);
+	if (!luaW_setwidgetcallback(L, &w, wd, "on_left_click")) {
+		connect_signal_notify_modified(w, std::bind(&dialog_callback, L, lua_ptr<gui2::widget>(w), "callback"));
+	}
+}
+
+WIDGET_SETTER("on_button_click", lua_index_raw, gui2::widget)
+{
+	gui2::window* wd = w.get_window();
+	gui2::clickable_item* cl = dynamic_cast<gui2::clickable_item*>(&w);
+
+	if(!wd) {
+		throw std::invalid_argument("the widget has no window assigned");
+	}
+	if(!cl) {
+		throw std::invalid_argument("unsupported widget");
+	}
+	lua_pushvalue(L, value.index);
+	if (!luaW_setwidgetcallback(L, &w, wd, "on_button_click")) {
+		cl->connect_click_handler(std::bind(&dialog_callback, L, lua_ptr<gui2::widget>(w), "on_button_click"));
+	}
+}
+
+}
+
+
+
+namespace lua_widget {
+
+int impl_widget_get(lua_State* L)
+{
+	gui2::widget& w = luaW_checkwidget(L, 1);
+	if(lua_isinteger(L, 2)) {
+		
+		if(auto pwidget = find_child_by_index(w, luaL_checkinteger(L, 2))) {
+			luaW_pushwidget(L, *pwidget);
+			return 1;
+		}
+		
+	}
+	utils::string_view str = lua_check<utils::string_view>(L, 2);
+
+	tgetters::iterator it = getters.find(std::string(str));
+	if(it != getters.end()) {
+		for(const auto& func : it->second) {
+			if(func(L, w)) {
+				return 1;
+			}
+		}
+	}
+	if(luaW_getglobal(L, "gui", "widget", std::string(str).c_str())) {
+		return 1;
+	}
+	if(auto pwidget = find_child_by_name(w, std::string(str))) {
+		luaW_pushwidget(L, *pwidget);
+		return 1;
+	}
+	ERR_LUA << "invalid propertly of '" <<  typeid(w).name()<< "' widget :" << str << "\n";
+	return luaL_argerror(L, 2, "invalid propertly of widget");
+}
+
+int impl_widget_set(lua_State* L)
+{
+	gui2::widget& w = luaW_checkwidget(L, 1);
+	utils::string_view str = lua_check<utils::string_view>(L, 2);
+
+
+	tsetters::iterator it = setters.find(std::string(str));
+	if(it != setters.end()) {
+		for(const auto& func : it->second) {
+			if(func(L, 3, w)) {
+				return 0;
+			}
+		}
+		ERR_LUA << "none of "<< it->second.size() << " setters macthed\n";
+	}
+	else {
+		ERR_LUA << "unknown peopertly id : " << str << " #known properties="  << setters.size() << "\n";
```

Comment:
This is just one of many typos in your code... it's one thing to have a typo in a comment, but you should really spell-check log messages and error messages at least.

Reply:
hm ye i guess i should fix them